src/components/LocationSelector/LocationSelectorWithGoogleLoader.js

```diff
@@ -0,0 +1,242 @@
+import LocationSelector from './LocationSelector';
+import { findCenter, getRadiusInMeters } from './utils';
+import styles from './LocationSelector.scss';
+
+const GOOGLE_API_LIBRARIES = ['places'];
+
+const { block, elem } = bem({
+    name: 'LocationSelector',
+    classnames: styles,
+    propsToMods: ['muted'],
+});
+
+function LocationSelectorWithGoogleLoader(props) {
+    const {
+        /** Google props */
+        apiKey,
+        language,
+        region,
+        additionalGoogleProps,
+
+        /** FieldWrapper props */
+        clearLabel,
+        onRemoveAllLocations,
+        inputPlaceholder,
+        selectionPlaceholder,
+
+        /** Modal props */
+        modalContentLabel,
+
+        /** LocationCard props */
+        minRadius,
+        maxRadius,
+        radiusStep,
+        renderRadiusLabel,
+        onRemoveLocation,
+        doneLabel,
+
+        /** LocationAutocomplete props */
+        country,
+        placeTypes,
+        noSuggestionsPlaceholder,
+        showCountryInSuggestions,
+        onLocationAutocompleteError,
+
+        /** Internal use */
+        onAddLocation,
+        onUpdateLocation,
+        onBlur,
+        selectedLocations,
+        radiusDefaultValue,
+        radiusUnits,
+        ...rest
+    } = props;
+
+    const [isOpen, setIsOpen] = React.useState(false);
+
+    function handleOpenModal() {
+        if (!isOpen) {
+            setIsOpen(true);
+        }
+    }
+
+    function handleCloseModal() {
+        if (isOpen) {
+            setIsOpen(false);
+            onBlur();
+        }
+    }
+
+    /**
+     * Fetch additional information for the selected place and
+     * add it along with passed location object to the selectedLocations array
+     */
+    function handleAddLocation(location) {
+        const { Geocoder } = window.google.maps;
+        const geocoder = new Geocoder();
+
+        findCenter(geocoder, location.place_id)
+            .then(center => {
+                const locationToAdd = {
+                    ...location,
+                    center,
+                    radius: radiusDefaultValue,
+                };
+                onAddLocation(locationToAdd);
+            })
+            .catch(/* TODO: add error handling */);
+    }
+
+    function getMarkers() {
+        return selectedLocations.map(location => ({
+            center: location.center,
+            radius: getRadiusInMeters(location.radius, radiusUnits),
+        }));
+    }
+
+    return (
+        <div {...block(props)}>
+            <FieldWrapper
+                clearLabel={clearLabel}
+                onClick={handleOpenModal}
+                onClear={onRemoveAllLocations}
+                {...rest}
+            >
+                <FaMapMarkerAlt {...elem('icon', props)} />
+                <Text {...elem('mainTextInput', { ...props, muted: !selectionPlaceholder })}>
+                    {selectionPlaceholder || inputPlaceholder}
+                </Text>
+            </FieldWrapper>
```

Comment:
Would be nice for more transparency create separate component like 'FaKeLocationInput'  and put this there.

Reply:
@mukiienko I think it's a quite unique case for now and it won't be reused anywhere else. Besides that I find it pretty self-descriptive, short so I'm not sure if we should do it right now